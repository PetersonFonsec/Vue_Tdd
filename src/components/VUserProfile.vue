<template>
    <div>
        VUserProfile
    </div>
</template>

<script>
export default {
    name: 'VUserProfile',
    props: {
        user: {
            type: Object,
            required: true,
            default: () => ({})
        }
    }
}
</script>

<style>

</style>
