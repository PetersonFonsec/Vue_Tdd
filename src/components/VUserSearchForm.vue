<template>
    <div>
        VUserSearchForm
    </div>
</template>

<script>
export default {
    name: 'VUserSearchForm'
}
</script>

<style>

</style>
