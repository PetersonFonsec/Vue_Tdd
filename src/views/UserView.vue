<template>
    <div>
        <VUserProfile :user="user" />
        <VUserSearchForm/>
        UserView
    </div>
</template>

<script>
import VUserProfile from '@/components/VUserProfile'
import VUserSearchForm from '@/components/VUserSearchForm'

export default {
    name : 'UserView',
    components: {
        VUserProfile,
        VUserSearchForm
    },
    data(){
        return{
            user: { name: '' }
        }
    }    
}
</script>

<style>

</style>
