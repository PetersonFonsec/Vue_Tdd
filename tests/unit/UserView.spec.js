import { shallowMount } from '@vue/test-utils'
import UserView from '@/views/UserView'

import VUserSearchForm from '@/components/VUserSearchForm'

import VUserProfile from '@/components/VUserProfile'

describe('UserView', ()=>{
    
    const build = () => {
        const wrapper = shallowMount(UserView, {
            data : () => ({
                user: {}
            })
        }) 

        return {
            wrapper,
        
            userSearchForm: () => wrapper.find(VUserSearchForm) ,
        
            userProfile: () => wrapper.find(VUserProfile)
        }
    }

    it('renders the component', () => {

        const { wrapper } = build()

        expect(wrapper.html()).toMatchSnapshot()

    })

    it('render main child components',()=>{
        
        const { userSearchForm, userProfile } = build()

        expect( userSearchForm().exists() ).toBe( true )
        
        expect( userProfile().exists() ).toBe( true )
    })

    it('passes a binded user prop to user profile component', () =>{

        const { wrapper, userProfile } = build()
        
        wrapper.setData({
            user:{
                name: 'Peterson'
            }
        })

        expect(userProfile().vm.user).toBe(wrapper.vm.user)
    })
})