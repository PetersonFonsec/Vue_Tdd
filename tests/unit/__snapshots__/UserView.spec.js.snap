// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView renders the component 1`] = `
<div>
  <vuserprofile-stub user="[object Object]"></vuserprofile-stub>
  <vusersearchform-stub></vusersearchform-stub>
  UserView
</div>
`;
